vectormath/vector.py

```diff
@@ -6,28 +6,15 @@
 import numpy as np
 
 
-class Vector3(np.ndarray):
-    """Primitive 3D vector defined from the origin"""
-
-    def __new__(cls, x=None, y=None, z=None):
-
-        def read_array(X, Y, Z):
-            if isinstance(X, cls) and Y is None and Z is None:
-                return cls(X.x, X.y, X.z)
-            if (isinstance(X, (list, tuple, np.ndarray)) and len(X) == 3 and
-                    Y is None and Z is None):
-                return cls(X[0], X[1], X[2])
-            if X is None and Y is None and Z is None:
-                return cls(0, 0, 0)
-            if np.isscalar(X) and np.isscalar(Y) and np.isscalar(Z):
-                xyz = np.r_[X, Y, Z]
-                xyz = xyz.astype(float)
-                return xyz.view(cls)
-            raise ValueError('Invalid input for Vector3 - must be an instance '
-                             'of a Vector3, a length-3 array, 3 scalars, or '
-                             'nothing for [0., 0., 0.]')
+class Vector(np.ndarray):
 
-        return read_array(x, y, z)
+    def __new__(cls, *args, **kwargs):
+        for vcls in (Vector3, Vector2):
```

Comment:
That isn't a typo - constructing Vector (`cls`) tries to construct either Vector3 or Vector2 (the `vcls`es). I think I'll just take it out though... It is unnecessary complication. `BaseVector` and `BaseVectorArray` should just not be used at all.